File: admin/src/api/status.ts
import { getJson } from './client';

export type HealthState = 'ready' | 'degraded' | 'unavailable' | 'unknown';
export type RuntimeState = 'starting' | 'ready' | 'degraded' | 'unavailable' | 'stopping';

export type HealthSnapshot = {
  state: HealthState;
  message?: string;
  hint?: string;
};

export type LocalStorageSnapshot = HealthSnapshot & {
  provider: 'Local';
  path?: string;
};

export type RuntimeStatus = {
  state: RuntimeState;
  observedAt: string;
  database: HealthSnapshot;
  localStorage: HealthSnapshot;
  projectSource?: string;
  version?: string;
};

export type StorageStatus = {
  database: HealthSnapshot;
  localStorage: LocalStorageSnapshot;
};

export function fetchRuntimeStatus(signal?: AbortSignal): Promise<RuntimeStatus> {
  return getJson<RuntimeStatus>('/admin/api/v1/runtime/status', signal);
}

export function fetchStorageStatus(signal?: AbortSignal): Promise<StorageStatus> {
  return getJson<StorageStatus>('/admin/api/v1/storage/status', signal);
}

File: internal/diagnostics/status.go
package diagnostics

import "time"

type Health struct {
	State   string
	Message string
	Hint    string
}

type RuntimeStatus struct {
	State         string
	ObservedAt    time.Time
	Database      Health
	LocalStorage  Health
	ProjectSource string
	Version       string
}

type LocalStorageStatus struct {
	State    string
	Message  string
	Hint     string
	Provider string
	Path     string
}

type FileStorageStatus struct {
	State          string
	Provider       string
	ActiveProvider string
	Message        string
	Hint           string
}

type StorageStatus struct {
	Database     Health
	LocalStorage LocalStorageStatus
	FileStorage  FileStorageStatus
}
